Editor/EditorCore/UVEditor.cs

```diff
@@ -718,44 +718,81 @@ void OnScreenResize()
             Repaint();
         }
 
+        bool IsCopyUVSettingsModifiers(EventModifiers modifiers)
+        {
+            return (modifiers == (EventModifiers.Control | EventModifiers.Shift));
+        }
+
         /**
-         * return true if shortcut should eat the event
+         * returns true if a copy of UV settingsoccured from the first selected face to the target face
          */
-        internal bool ClickShortcutCheck(ProBuilderMesh pb, Face selectedFace)
+        bool CopyFaceUVSettings(ProBuilderMesh pb, Face targetFace)
         {
-            Event e = Event.current;
+            // get first selected UV face
+            ProBuilderMesh firstObj;
+            Face[] source = new Face[1];
 
-            // Copy UV settings
-            if (e.modifiers == (EventModifiers.Control | EventModifiers.Shift))
+            ProBuilderEditor.instance.GetFirstSelectedFace(out firstObj, out source[0]);
+
+            if (source != null)
             {
-                // get first selected Auto UV face
-                ProBuilderMesh firstObj;
-                Face source;
+                UndoUtility.RecordObject(pb, "Copy UV Settings");
 
-                ProBuilderEditor.instance.GetFirstSelectedFace(out firstObj, out source);
+                bool destinationWasManualUV = targetFace.manualUV;
+                bool sourceWasManualUV = source[0].manualUV;
+                if (sourceWasManualUV == true)
+                {
+                    //We need to convert it to auto
+                    UVEditing.SetAutoUV(firstObj, source, true);
+                    pb.Refresh();
+                    pb.Optimize();
+                }
+
+                targetFace.uv = new AutoUnwrapSettings(source[0].uv);
+                targetFace.submeshIndex = source[0].submeshIndex;
+                EditorUtility.ShowNotification("Copy UV Settings");
 
-                if (source != null)
+                if (sourceWasManualUV == true)
                 {
-                    UndoUtility.RecordObject(pb, "Copy UV Settings");
+                    //We need to convert it back to false
+                    UVEditing.SetAutoUV(firstObj, source, false); 
+                }
 
-                    selectedFace.uv = new AutoUnwrapSettings(source.uv);
-                    selectedFace.submeshIndex = source.submeshIndex;
-                    EditorUtility.ShowNotification("Copy UV Settings");
+                pb.ToMesh();
+                pb.Refresh();
+                pb.Optimize();
```

Comment:
Same note as above, we only need these calls at the end of mesh edits.

Reply:
Same comment as above